include/seqan/basic/alphabet_math.h

```diff
@@ -220,26 +220,26 @@ template <typename T> inline T const & supremumValueImpl(T *);
 // Forward to supremumValueImpl() only.
 
 template <typename T>
-inline T const &
+[[deprecated("Use MaxValue instead.")]]
+inline T
 maxValue()
 {
-    T * _tag = 0;
-    return supremumValueImpl(_tag);
+    return MaxValue<T>::VALUE;
```

Comment:
directly use the numeric_limits and also recommend them in the deprecation notice